clients/typescript/src/satellite/process.ts

```diff
@@ -404,6 +433,10 @@ export class SatelliteProcess implements Satellite {
     // this could especially happen in unit tests
     this.subscriptionNotifiers[subId] = emptyPromise()
 
+    if (this.initializing) {
+      await this.initializing.promise
+    }
```

Comment:
```suggestion
    await this.initializing?.promise
```

Awaiting undefined is a no-op, so we can do it like that.